File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string str=to_string(x);
        int n=str.length();
        int i=0;
        int j=n-1;
        for(int i=0;i<j;i++)
        {
            if(str[i]!=str[j])
            {
                return false;
            }
            j--;
        }
        return true;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x):
        if x < 0: 
            return False

        p1 = x
        counter = 0 

        if x%10==0 and x//10!=0:
            return False
        while x>counter:
            counter = (counter * 10) + (x % 10)
            if x!=counter:
                x = x // 10
        return x == counter
        